### PR TITLE
EDK2-Test: Bug 4244 - SCT improvement - Print out the PCIe device pat…

### PR DESCRIPTION
**Pull request by PrachotanReddy**
https://bugzilla.tianocore.org/show_bug.cgi?id=4244
Tests Modified:
AdapterInfoBBTestConformance
HIIConfigAccessBBTestConformance
PxeBaseCodeBBTestConformance
SimpleNetworkBBTestConformance

UEFI-SCT
Cc: G Edhaya Chandran <Edhaya.Chandran@arm.com>
Cc: Barton Gao <gaojie@byosoft.com.cn>
Cc: Carolyn Gjertsen <Carolyn.Gjertsen@amd.com>
Cc: Samer El-Haj-Mahmoud <Samer.El-Haj-Mahmoud@arm.com>
Cc: Eric Jin <eric.jin@intel.com>
Cc: Arvin Chen <arvinx.chen@intel.com>
Cc: Supreeth Venkatesh <Supreeth.Venkatesh@amd.com>

### CODE REVIEW

**Comment by PrachotanReddy** [comment]
Here's snippets of output from a run on Ampere Altra Developer Platform

**AdapterInfoBBTestConformance**

```
------------------------------------------------------------
Conformance auto test for Adapter Information Protocol GetInformation().
------------------------------------------------------------
Logfile: "\AARCH64\Overall\Summary.log"
Test Started: 05/31/24  09:39p
-----------------------------------------------------------
Device Path: Acpi(PNP0A08,4)/Pci(1|0)/Pci(0|0)/Mac(00306470AFB6)
```

**HIIConfigAccessBBTestConformance**

```
------------------------------------------------------------
Conformance auto test for HII Configuration Access Protocol ExtractConfig().
------------------------------------------------------------
Logfile: "\AARCH64\Log\HIITest\HIIConfigAccessProtocolTest0\ExtractConfigConform
ance_15_0_E2C3B0FE-FBE9-46A9-A31B-A3F3820DCF04.log"
Test Started: 05/31/24  07:14p
------------------------------------------------------------
Device Path: Acpi(PNP0A08,4)/Pci(1|0)/Pci(0|0)/Mac(00306470AFB6)/VenHw(9FB1A1F3-
3B71-4324-B39A-745CBB015FFF)
```

**PxeBaseCodeBBTestConformance**
```
------------------------------------------------------------
Auto Conformance Test for Arp
------------------------------------------------------------
Logfile: "\AARCH64\Log\NetworkSupportTest\PXEBaseCodeProtocolTest0\Arp_Conf_0_0_
4C89BAC6-21EA-4F04-A2B6-EF67D34F6498.log"
Test Started: 05/31/24  07:11p
------------------------------------------------------------
Device Path: Acpi(PNP0A08,4)/Pci(1|0)/Pci(0|0)/Mac(00306470AFB6)/IPv4(0.0.0.0:0)
```

**SimpleNetworkBBTestConformance**
```
------------------------------------------------------------
Conformance Test for GetStatus
------------------------------------------------------------
Logfile: "\AARCH64\Log\NetworkSupportTest\SimpleNetworkProtocolTest0\GetStatus_C
onf_7_0_574C0B74-ADC3-46D9-A8FF-C8A44E6646FA.log"
Test Started: 05/31/24  06:57p
------------------------------------------------------------
Device Path: Acpi(PNP0A08,4)/Pci(1|0)/Pci(0|2)/Mac(00306470AFB8)
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c [hunk 1]
```diff
@@ -207,3 +207,99 @@ UnloadAdapterInfoBBTest (
...
+  if (Index >= NoHandles)
+  {
+    //
+    // No Handle Found!!
+    //
+    gtBS->FreePool(HandleBuffer);
+    return EFI_DEVICE_ERROR;
+  }
+
+  Status = gtBS->HandleProtocol(
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c:298]:
```suggestion
  Status = gtBS->HandleProtocol (

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestConformance.c [hunk 2]
```diff
@@ -137,6 +138,28 @@ BBTestExtractConfigConformanceTest (
...
+      StandardLib->RecordMessage (
+                     StandardLib,
+                     EFI_VERBOSE_LEVEL_DEFAULT,
+                     L"Device Path: %s\r\n",
+                     DevicePathStr
+                     );
+      Status = gtBS->FreePool (DevicePathStr);
+      if (EFI_ERROR(Status))
+        return Status;
+      DevicePathStr=NULL;
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestConformance.c:160]: Same coding style related comment as we need to follow https://tianocore-docs.github.io/edk2-CCodingStandardsSpecification/draft/5_source_files/52_spacing.html#52-spacing.

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.h [hunk 3]
```diff
@@ -160,12 +162,9 @@ BBTestGetSupportedTypesFunctionTest (
...
-  IN EFI_HANDLE                 ImageHandle,
-  IN EFI_SYSTEM_TABLE           *SystemTable
-  );
-
-
+EFI_STATUS
+LocateDevicePathFromAdapterInfo(
+  IN EFI_ADAPTER_INFORMATION_PROTOCOL *AdapterInfo2,
+  IN EFI_DEVICE_PATH_PROTOCOL **DevicePath,
+  IN EFI_STANDARD_TEST_LIBRARY_PROTOCOL *StandardLib);
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.h:169]:
```suggestion
  IN EFI_STANDARD_TEST_LIBRARY_PROTOCOL *StandardLib
  );

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c [hunk 4]
```diff
@@ -381,6 +381,71 @@ GetDevicePath (
   return Status;
 }
 
+EFI_STATUS
+GetDevicePathProtocol (
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c:385]: Why don't we make a similar change to HiiConfigAccess protocol test? having a function called LocateDevicePathFromHIIConfigAccess ()

**Review by sunnywang-arm — COMMENTED** [review]
Thanks for working on this, Prachotan. 
Most issues are the coding style related issues as we need to follow https://tianocore-docs.github.io/edk2-CCodingStandardsSpecification/draft/5_source_files/52_spacing.html#52-spacing. I only added comments for the changes in AdapterInfo/BlackBoxTest for your reference. Please also fix the issues with other files. 

Could you also add the code block to the tests below? I'm fine with you sending the other patch for the tests below if you don't want to have a big change. 
- BBTestGetInformationFunctionTest
- BBTestSetInformationFunctionTest
- BBTestGetSupportedTypesFunctionTest
- BBTestExtractConfigFunctionTest
- BBTestRouteConfigFunctionTest
- BBTestNewStartFunctionTest
- BBTestNewStopFunctionTest
- BBTestSetIpFilterFunctionTest
- BBTestStartFunctionTest
- BBTestStopFunctionTest
- BBTestInitializeFunctionTest
- BBTestResetFunctionTest
- BBTestShutdownFunctionTest
- BBTestReceiveFilterFunctionTest
- BBTestStationAddressFunctionTest
- BBTestStatisticsFunctionTest
- BBTestMCastIpToMacFunctionTest
- BBTestNVDataFunctionTest
- BBTestGetStatusFunctionTest

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c [hunk 5]
```diff
@@ -207,3 +207,110 @@ UnloadAdapterInfoBBTest (
...
+        L"BS.LocateHandle - LocateHandle",
+        L"%a:%d:Status - %r",
+        __FILE__,
+        (UINTN)__LINE__,
+        Status);
+    return Status;
+  }
+
+  if (NoHandles <= 0)
+  {
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c:249]:
```suggestion
  if (NoHandles <= 0) {

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c [hunk 6]
```diff
@@ -207,3 +207,110 @@ UnloadAdapterInfoBBTest (
...
+    StandardLib->RecordAssertion(
+        StandardLib,
+        EFI_TEST_ASSERTION_FAILED,
+        gTestGenericFailureGuid,
+        L"BS.LocateHandle - LocateHandle",
+        L"%a:%d:Device Error",
+        __FILE__,
+        (UINTN)__LINE__);
+    return EFI_DEVICE_ERROR;
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c:258]:
```suggestion
    return EFI_UNSUPPORTED;

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c [hunk 7]
```diff
@@ -207,3 +207,110 @@ UnloadAdapterInfoBBTest (
...
+      break;
+    }
+
+  }
+
+  //
+  // Locate the Loaded DevicePath Protocol bound to ADAPTER_INFORMATION Protocol
+  //
+  if (Index >= NoHandles)
+  {
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c:298]:
```suggestion
  if (Index >= NoHandles) {

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c [hunk 8]
```diff
@@ -336,3 +336,103 @@ BBTestSimpleNetworkUnload (
...
+        L"%a:%d:Status - %r",
+        __FILE__,
+        (UINTN)__LINE__,
+        Status);
+    return Status;
+  }
+
+  if (NoHandles <= 0)
+  {
+    StandardLib->RecordAssertion(
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c:378]:
```suggestion
  if (NoHandles <= 0) {
    StandardLib->RecordAssertion(
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c [hunk 9]
```diff
@@ -336,3 +336,103 @@ BBTestSimpleNetworkUnload (
...
+        StandardLib,
+        EFI_TEST_ASSERTION_FAILED,
+        gTestGenericFailureGuid,
+        L"BS.LocateHandle - LocateHandle",
+        L"%a:%d:Device Error",
+        __FILE__,
+        (UINTN)__LINE__);
+    return EFI_DEVICE_ERROR;
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c:386]: EFI_NOT_FOUND or EFI_UNSUPPORTED

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c [hunk 10]
```diff
@@ -336,3 +336,103 @@ BBTestSimpleNetworkUnload (
...
+          __FILE__,
+          (UINTN)__LINE__,
+          Status);
+
+      gtBS->FreePool(HandleBuffer);
+      return Status;
+    }
+
+    if (OtherSnpInterface == SnpInterface)
+    {
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c:414]:
```suggestion
    if (OtherSnpInterface == SnpInterface) {
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c [hunk 11]
```diff
@@ -336,3 +336,103 @@ BBTestSimpleNetworkUnload (
...
+  //
+  // Find the exact handle that SnpInterface bound to
+  //
+  for (Index = 0; Index < NoHandles; Index++) {
+    Status = gtBS->HandleProtocol(
+        HandleBuffer[Index],
+        &gBlackBoxEfiSimpleNetworkProtocolGuid,
+        (VOID **) &OtherSnpInterface);
+    if (EFI_ERROR(Status))
+    {
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c:398]:
```suggestion
    if (EFI_ERROR(Status)) {
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c [hunk 12]
```diff
@@ -336,3 +336,103 @@ BBTestSimpleNetworkUnload (
...
+      break;
+    }
+  }
+
+  //
+  // Locate the Loaded DevicePath Protocol bound to SIMPLE_NETWORK Protocol
+  //
+  if (Index >= NoHandles)
+  {
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c:423]:
```suggestion
  if (Index >= NoHandles) {
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c [hunk 13]
```diff
@@ -336,3 +336,103 @@ BBTestSimpleNetworkUnload (
...
+    //
+    gtBS->FreePool(HandleBuffer);
+    return EFI_DEVICE_ERROR;
+  }
+
+  Status = gtBS->HandleProtocol(
+      HandleBuffer[Index],
+      &gBlackBoxEfiDevicePathProtocolGuid,
+      (VOID **) DevicePath);
+
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c:435]:
```suggestion
      (VOID **) DevicePath
      );

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c [hunk 14]
```diff
@@ -336,3 +336,103 @@ BBTestSimpleNetworkUnload (
...
+  //
+  // Locate the Loaded DevicePath Protocol bound to SIMPLE_NETWORK Protocol
+  //
+  if (Index >= NoHandles)
+  {
+    //
+    // No Handle Found!!
+    //
+    gtBS->FreePool(HandleBuffer);
+    return EFI_DEVICE_ERROR;
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestMain.c:428]: EFI_NOT_FOUND  or EFI_UNSUPPORTED

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c [hunk 15]
```diff
@@ -957,149 +742,255 @@ SetIpAddress (
...
+  //
+  // Locate the Loaded Image DevicePath Protocol bound to PXE_BASE_CODE Protocol
+  //
+  if (Index >= NoHandles)
+  {
+    //
+    // No Handle Found!!
+    //
+    gtBS->FreePool(HandleBuffer);
+    return EFI_DEVICE_ERROR;
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c:986]: EFI_NOT_FOUND or EFI_UNSUPPORTED

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c [hunk 16]
```diff
@@ -957,149 +742,255 @@ SetIpAddress (
...
+      break;
+    }
+  }
+
+  //
+  // Locate the Loaded Image DevicePath Protocol bound to PXE_BASE_CODE Protocol
+  //
+  if (Index >= NoHandles)
+  {
+    //
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c:982]:
```suggestion
  if (Index >= NoHandles) {
    //
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c [hunk 17]
```diff
@@ -957,149 +742,255 @@ SetIpAddress (
...
+  {
+    StandardLib->RecordAssertion(
+        StandardLib,
+        EFI_TEST_ASSERTION_FAILED,
+        gTestGenericFailureGuid,
+        L"BS.LocateHandle - LocateHandle",
+        L"%a:%d:Device Error",
+        __FILE__,
+        (UINTN)__LINE__);
+    return EFI_DEVICE_ERROR;
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c:944]: EFI_NOT_FOUND or EFI_UNSUPPORTED

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c [hunk 18]
```diff
@@ -957,149 +742,255 @@ SetIpAddress (
...
+        L"BS.LocateHandle - LocateHandle",
+        L"%a:%d:Status - %r",
+        __FILE__,
+        (UINTN)__LINE__,
+        Status);
+    return Status;
+  }
+
+  if (NoHandles <= 0)
+  {
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c:935]:
```suggestion
  if (NoHandles <= 0) {
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c [hunk 19]
```diff
@@ -957,149 +742,255 @@ SetIpAddress (
...
+  // Locate the Handle that the BcInterface is bound to
+  //
+  Status = gtBS->LocateHandleBuffer(
+      ByProtocol,
+      &gBlackBoxEfiPxeBaseCodeProtocolGuid,
+      NULL,
+      &NoHandles,
+      &HandleBuffer);
+  if (EFI_ERROR(Status))
+  {
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c:921]:
```suggestion
  if (EFI_ERROR(Status)) {
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c [hunk 20]
```diff
@@ -957,149 +742,255 @@ SetIpAddress (
...
+  gtBS->FreePool(TempBuffer);
   return EFI_SUCCESS;
 }
+
+EFI_STATUS
+LocateDevicePathFromBcInterface(
+  IN EFI_PXE_BASE_CODE_PROTOCOL *BcInterface,
+  IN EFI_DEVICE_PATH_PROTOCOL **DevicePath,
+  IN EFI_STANDARD_TEST_LIBRARY_PROTOCOL *StandardLib)
+{
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/PxeBaseCodeBBTestMain.c:904]:
```suggestion
  IN EFI_STANDARD_TEST_LIBRARY_PROTOCOL *StandardLib
  )
{
```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c [hunk 21]
```diff
@@ -207,3 +207,110 @@ UnloadAdapterInfoBBTest (
...
+    //
+    // No Handle Found!!
+    //
+    gtBS->FreePool(HandleBuffer);
+    return EFI_DEVICE_ERROR;
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestMain.c:303]: EFI_NOT_FOUND or EFI_UNSUPPORTED

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestConformance.c [hunk 22]
```diff
@@ -111,6 +112,37 @@ BBTestGetInformationConformanceTest (
     return Status;
   }
 
+  Status = LocateDevicePathFromAdapterInfo(AdapterInfo,&DevicePath,StandardLib);
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestConformance.c:115]:
```suggestion
  Status = LocateDevicePathFromAdapterInfo (AdapterInfo, &DevicePath, StandardLib);

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestConformance.c [hunk 23]
```diff
@@ -155,6 +188,35 @@ BBTestSetInformationConformanceTest (
...
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestConformance.c:191]:
```suggestion
  Status = LocateDevicePathFromAdapterInfo (AdapterInfo, &DevicePath, StandardLib);

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestConformance.c [hunk 24]
```diff
@@ -195,7 +258,24 @@ BBTestGetSupportedTypesConformanceTest (
   if ( EFI_ERROR(Status) ) {
     return Status;
   }
-
+  Status = LocateDevicePathFromAdapterInfo(AdapterInfo,&DevicePath,StandardLib);
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/AdapterInfo/BlackBoxTest/AdapterInfoBBTestConformance.c:261]:
```suggestion
  Status = LocateDevicePathFromAdapterInfo (AdapterInfo, &DevicePath, StandardLib);

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestConformance.c [hunk 25]
```diff
@@ -72,6 +74,24 @@ BBTestStartConformanceTest (
   //
   SnpInterface = (EFI_SIMPLE_NETWORK_PROTOCOL *)ClientInterface;
 
+  Status = LocateDevicePathFromSnpInterface(SnpInterface,&DevicePath,StandardLib);
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestConformance.c:77]:
```suggestion
  Status = LocateDevicePathFromSnpInterface (SnpInterface, &DevicePath, StandardLib);

```

File: uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestConformance.c [hunk 26]
```diff
@@ -1007,6 +1142,24 @@ BBTestStatisticsConformanceTest (
...
+                     StandardLib,
+                     EFI_VERBOSE_LEVEL_DEFAULT,
+                     L"Device Path: %s\r\n",
+                     DevicePathStr
+                     );
+      Status = gtBS->FreePool (DevicePathStr);
+      if (EFI_ERROR(Status)) {
+        return Status;
+      }
+      DevicePathStr=NULL;
```
Comment (sunnywang-arm) [inline uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleNetwork/BlackBoxTest/SimpleNetworkBBTestConformance.c:1160]:
```suggestion
      DevicePathStr = NULL;

```

**Review by sunnywang-arm — CHANGES_REQUESTED** [review]
A few more minor comments for cleanup. Others look good to me.

**Pull request closed without merging**